src/lib/hash/sha2_64/sha2_64.cpp

```diff
@@ -29,23 +30,11 @@ std::string sha512_provider() {
 
 }  // namespace
 
-std::unique_ptr<HashFunction> SHA_384::copy_state() const {
-   return std::make_unique<SHA_384>(*this);
-}
-
-std::unique_ptr<HashFunction> SHA_512::copy_state() const {
-   return std::make_unique<SHA_512>(*this);
-}
-
-std::unique_ptr<HashFunction> SHA_512_256::copy_state() const {
-   return std::make_unique<SHA_512_256>(*this);
-}
```

Comment:
I think the various `copy_state` and such should *not* be inlined; it's already an expensive operation due to requiring a heap allocation.

In the predecessor PR I actually went in the other direction and moved all of the `new_object` to be not inline.